### ISSUE TITLE
Supress domains not working on certain domains?

### ISSUE THREAD

**Issue by duramato** [quidsup/notrack#143]
As seen in the screenshoot Nº1 `er.spotify.com`is added to supress domains but it isn't surpressed as seen by screenshoot Nº2

**1:**
![image](https://cloud.githubusercontent.com/assets/10121067/18997087/d4e558d0-872a-11e6-9047-472189bb520d.png)

**2:**
![image](https://cloud.githubusercontent.com/assets/10121067/18997095/dd2498c6-872a-11e6-91e8-f7136a626710.png)

**Comment by quidsup** [comment]
The feature works by matching from the right hand side of the URL and it only focuses on domain and top level domain. 
In order to suppress spotify, you need to drop the sub-domain `er.` and write it as: `spotify.com`

**Comment by quidsup** [comment]
Need to add server side checking to ensure domains are inputted correctly

**Comment by duramato** [comment]
Mmm I would like to have subdomain suppressing too, as there are some I don't the join all of the TLD. Would that be something possible to have?

**Comment by quidsup** [comment]
Yes it can be done, however it adds to the processing requirements.

This is the function in question, which gets called against every single DNS request.
```
*  Simplify URL
 *  
 *  1: Drop www (its unnecessary and not all websites use it now)
 *  2. Extract domain.tld, including double-barrelled domains
 *  3. Check if site is to be suppressed (present in $CommonSites)
 *
 *  Params:
 *    $url - URL To Simplify
 *  Return:
 *    Simplified URL
 */
function simplify_url($url) {  
  global $CommonSites;
  $simpleurl = '';
    
  if (substr($url,0,4) == 'www.') $simpleurl  = substr($url,4); 
  else $simpleurl  = $url;
  
  if (preg_match('/[A-Za-z0-9\-\_]{2,63}\.(gov\.|org\.|co\.|com\.)?[A-Za-z0-9\-]{2,63}$/', $simpleurl , $Match) == 1) {
    if (in_array($Match[0],$CommonSites)) return '*.'.$Match[0];
    else return $simpleurl ;
  }
 
  return $simpleurl ;
}
```

The minimum I can see is adding another group check to the regex, and another if statement. However, Its too much for an RPi 2 after a busy Internet session.
If the move over to using a SQL #141 give enough of a speed boost, then maybe it will be feasable to add your request, but right now I have to balance the exra requirements against the low power devices.
